exporter/clickhousemetricsexporterv2/exporter.go

```diff
@@ -291,52 +251,51 @@ func (c *clickhouseMetricsExporter) processGauge(batch *writeBatch, metric pmetr
 			value = dp.DoubleValue()
 		}
 		unixMilli := dp.Timestamp().AsTime().UnixMilli()
-		pointAttrs := dp.Attributes()
-		batch.samples = append(batch.samples, sample{
+
+		fingerprint := internal.NewFingerprint(internal.PointFingerprintType, scopeFingerprint.Hash(), dp.Attributes(), map[string]string{})
+		batch.addSample(&sample{
 			env:         env,
 			temporality: temporality,
 			metricName:  name,
-			fingerprint: Fingerprint(pointAttrs, scopeAttrs, resAttrs, name),
+			fingerprint: fingerprint.HashWithName(name),
 			unixMilli:   unixMilli,
 			value:       value,
 		})
-		c.processMetadata(batch, name, desc, unit, typ, temporality, isMonotonic, pointAttrs, pointAttrType)
-
-		batch.ts = append(batch.ts, ts{
+		batch.addMetadata(name, desc, unit, typ, temporality, isMonotonic, fingerprint)
+		batch.addTs(&ts{
 			env:           env,
 			temporality:   temporality,
 			metricName:    name,
 			description:   desc,
 			unit:          unit,
 			typ:           typ,
 			isMonotonic:   isMonotonic,
-			fingerprint:   Fingerprint(pointAttrs, scopeAttrs, resAttrs, name),
+			fingerprint:   fingerprint.HashWithName(name),
 			unixMilli:     unixMilli,
-			labels:        getJSONString(getAllLabels(pointAttrs, scopeAttrs, resAttrs, name)),
-			attrs:         getAttrMap(pointAttrs),
-			scopeAttrs:    getAttrMap(scopeAttrs),
-			resourceAttrs: getAttrMap(resAttrs),
+			labels:        internal.NewLabels(name, fingerprint.AttributesAsMap(), scopeFingerprintMap, resourceFingerprintMap),
```

Comment:
I would keep the name as `getJSONString`. The `NewLabels` returns a string only be used for writing and is not usable beyond the part of `ts`. The `getJSONString` is more clear what it is doing.